src/modules/loops/while_loop.rs

```diff
@@ -0,0 +1,66 @@
+use heraclitus_compiler::prelude::*;
+use crate::fragments;
+use crate::modules::prelude::*;
+use crate::utils::context::Context;
+use crate::modules::block::Block;
+use crate::modules::expression::expr::Expr;
+use crate::modules::types::{Type, Typed};
+
+#[derive(Debug, Clone)]
+pub struct WhileLoop {
+    condition: Expr,
+    block: Block,
+}
+
+impl SyntaxModule<ParserMetadata> for WhileLoop {
+    syntax_name!("While Loop");
+
+    fn new() -> Self {
+        WhileLoop {
+            condition: Expr::new(),
+            block: Block::new().with_needs_noop().with_condition(),
+        }
+    }
+
+    fn parse(&mut self, meta: &mut ParserMetadata) -> SyntaxResult {
+        token(meta, "while")?;
+        // Parse the condition expression
+        let tok = meta.get_current_token();
+        syntax(meta, &mut self.condition)?;
+        
+        // Validate that the condition is a boolean expression
+        if self.condition.get_type() != Type::Bool {
+            return error!(meta, tok, format!(
+                "Expected boolean expression in while condition, got {}",
+                self.condition.get_type()
+            ));
+        }
+
+        token(meta, "{")?;
```

Comment:
Is single statement not supported for loops by design, not like `if` statement? Should we add this on the https://docs.amber-lang.com/basic_syntax/loops?